scalabel/eval/result.py

```diff
@@ -0,0 +1,167 @@
+            exclude (set[str]): Optional, the metrics not to convert
+        Returns:
+            data_frame (pandas.DataFrmae): the exported DataFrame
+        """
+        frame_dict: Dict[str, Scores] = defaultdict(dict)
+        for metric, scores_list in self.dict(
+            include=include, exclude=exclude  # type: ignore
+        ).items():
+            if not isinstance(scores_list, list):
+                continue
+            for scores in scores_list:
+                for cls_, score in scores.items():
+                    frame_dict[metric][cls_] = score
+        return DataFrame.from_dict(frame_dict)
+
+    def table(
+        self,
+        include: Optional[AbstractSet[str]] = None,
+        exclude: Optional[AbstractSet[str]] = None,
+    ) -> str:
+        """Convert data model into a table for formatted printing.
+
+        Args:
+            include (set[str]): Optional, the metrics to convert
+            exclude (set[str]): Optional, the metrics not to convert
+        Returns:
+            table (str): the exported table string
+        """
+        data_frame = self.pd_frame(include, exclude)
+        if not self._formatters:
+            formatters = {
+                metric: "{:.1f}".format for metric in data_frame.columns
+            }
+        else:
+            formatters = self._formatters
+
+        summary = data_frame.to_string(formatters=formatters)
+        strs = summary.split("\n")
+        split_line = "-" * len(strs[0])
+
+        for row_ind in self._row_breaks:
+            strs.insert(row_ind, split_line)
+        summary = "".join([f"{s}\n" for s in strs])
+        summary = "\n" + summary
+        return summary  # type: ignore
+
+    def __str__(self) -> str:
+        """Convert the data into a printable string."""
+        return self.table()
+
+    def summary(
+        self,
+        include: Optional[AbstractSet[str]] = None,
+        exclude: Optional[AbstractSet[str]] = None,
+    ) -> Scores:
+        """Convert the data into a flattened dict as the summary.
+
+        This function is different to the `.dict()` function.
+        As a comparison, `.dict()` will export all data fields as a nested
+        dict, While `.summary()` only exports most important information,
+        like the overall scores, as a flattened compact dict.
+
+        Args:
+            include (set[str]): Optional, the metrics to convert
+            exclude (set[str]): Optional, the metrics not to convert
+        Returns:
+            dict[str, int | float]: returned summary of the result
+        """
+        summary_dict: Dict[str, Union[int, float]] = dict()
+        for metric, scores_list in self.dict(
+            include=include, exclude=exclude  # type: ignore
+        ).items():
+            if not isinstance(scores_list, list):
+                summary_dict[metric] = scores_list
+            else:
+                summary_dict[metric] = scores_list[-1].get(
```

Comment:
why still work with OVERALL / AVERAGE here? We can just print all metrics in the top-level of the hierarchy? Also please specify how the List reflects the hierarchy (bottom-up or top-down?)